src/functions/retroGen.ts

```diff
@@ -1,49 +1,48 @@
-import { unmarshall } from "@aws-sdk/util-dynamodb";
 import { LambdaClient } from "@aws-sdk/client-lambda";
+import { unmarshall } from "@aws-sdk/util-dynamodb";
+import { SQSBatchItemFailure, SQSBatchResponse } from "aws-lambda";
 import * as rp from "request-promise";
 import { ERRORS } from "../assets/Enum";
+import { credentials } from "../handler";
+import { ActivitiesService } from "../services/ActivitiesService";
+import { LambdaService } from "../services/LambdaService";
 import { RetroGenerationService } from "../services/RetroGenerationService";
 import { SharePointAuthenticationService } from "../services/SharePointAuthenticationService";
 import { SharePointService } from "../services/SharePointService";
 import { TestResultsService } from "../services/TestResultsService";
-import { ActivitiesService } from "../services/ActivitiesService";
-import { LambdaService } from "../services/LambdaService";
-import { PutObjectCommandOutput } from "@aws-sdk/client-s3";
-import { credentials } from "../handler";
 
 /**
  * λ function to process a DynamoDB stream of test results into a queue for certificate generation.
- * @param event - DynamoDB Stream event
+ * @param event - SQS event that contains a DynamoDB stream event to parse out
  */
-const retroGen = async (event: any): Promise<void | PutObjectCommandOutput[]> => {
+const retroGen = async (event: any): Promise<SQSBatchResponse> => {
```

Comment:
Presumably the `event` here should be of type `SQSEvent`?

Reply:
yeah, lazy, updated that now, will commit and push later in case further feedback